### PR TITLE
test(client): Test default selection contents

### PR DESCRIPTION
**Pull request by SevInf**
Should help to prevent issues like #18148

### CODE REVIEW

File: packages/client/tests/functional/default-selection/tests.ts [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+    if (provider !== 'sqlite' && provider !== 'sqlserver') {
+      // @ts-test-if: provider !== 'sqlite' && provider !== 'sqlserver'
+      input.enum = 'A'
+    }
+
+    if (provider === 'postgresql' || provider === 'cockroachdb' || provider === 'mongodb') {
+      // @ts-test-if: provider === 'postgresql' || provider === 'cockroachdb' || provider === 'mongodb'
+      input.list = ['Hello', 'world']
+      // @ts-test-if: provider === 'postgresql' || provider === 'cockroachdb' || provider === 'mongodb'
```
Comment (aqrln) [inline packages/client/tests/functional/default-selection/tests.ts:26]: would it be worth it to extract this to a variable (e.g. `supportsScalarLists`)?

**Review by aqrln — APPROVED** [review]

File: packages/client/tests/functional/default-selection/tests.ts (hunk 1 above)
Comment (jkomyno) [reply packages/client/tests/functional/default-selection/tests.ts:26]: I agree with @aqrln. Let's turn line 23 into something like `if (supportsScalarsList.includes(['postgresql', 'cockroachdb', 'mongodb'])) { /* */ }`

Comment (SevInf) [reply packages/client/tests/functional/default-selection/tests.ts:26]: Variables won't work for `@ts-test-if` - it is evaluated before test is run using  matrix only. So they'll still be duplicated for type tests. With that in mind, would you still prefer variable at least for runtime conditions?

Comment (aqrln) [reply packages/client/tests/functional/default-selection/tests.ts:26]: > With that in mind, would you still prefer variable at least for runtime conditions?

Either is fine with me if the runtime condition is duplicated too, but I'd probably prefer it the way it currently is then.

**Pull request merged**